src/braket/circuits/free_parameter_expression.py

```diff
@@ -0,0 +1,122 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+
+from typing import Dict
+
+from sympy import Expr, sympify
+
+
+class FreeParameterExpression:
+    """
+    Class 'FreeParameterExpression'
+    """
+
+    def __init__(self, expression: Expr):
+        """
+                Initializes a FreeParameterExpression. Best practice is to initialize using
+                FreeParameters and Numbers. Not meant to be initialized directly.
```

Comment:
Why is it not meant to be initialized directly?

Reply:
Rather than initialize with a Sympy expression, we want `FreeParameterExpression`s to be built up algebraically from other `FreeParameterExpression`s, with `FreeParameter`s at the bottom.